runtime/src/main/cpp/utf8/core.h

```diff
@@ -89,10 +89,16 @@ namespace internal
         return (cp >= LEAD_SURROGATE_MIN && cp <= TRAIL_SURROGATE_MAX);
     }
 
+    template <typename u32>
```

Comment:
What's the point of this change?

Reply:
Do you mean the entire file change or this particular function? I believe it's extracted to reuse this check two times.

Reply:
About the changes in this file: https://github.com/JetBrains/kotlin-native/pull/2950/files#r282302535